### ISSUE TITLE
Enhancement: add `readonly` property to calcite-input

### ISSUE THREAD

**Issue by caripizza** [Esri/calcite-components#2726]
### Description
Add `readonly` property to calcite-input, ensure it renders with accompanying styles. Per @macandcheese:

>we used to spread any attributes on the input to the rendered native input, so adding readonly would work and the referenced styles would be applied. But I think that was removed.

See [Discussion in Teams](https://teams.microsoft.com/l/message/19:fd15b51dacd24e70895ec1218a54ae06@thread.skype/1628092970609?tenantId=aee6e3c9-711e-4c7c-bd27-04f2307db20d&groupId=56fae21a-9407-4943-859f-a9bfcf0bbad3&parentMessageId=1628092970609&teamName=Calcite%20Design%20System&channelName=Calcite%20Components&createdTime=1628092970609) for additional info/context.

### Acceptance Criteria 
When calcite-input renders with a `readonly` attribute, it should appear with the corresponding "Read-only" styles per Figma.

### Relevant Info 
Ionic reference: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L160
MDN docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-readonly

#### Which Component
calcite-input

#### Example Use Case
Per Figma designs:
![image](https://user-images.githubusercontent.com/42423180/128226076-dc2a88b8-fb94-4166-99e4-a9d7dda9b46f.png)

**Comment by caripizza** [comment]
@macandcheese please add anything I missed here. @julio8a @jcfranco we may need to add this one to the current sprint.

**Comment by macandcheese** [comment]
Thanks for adding. Probably a larger discussion around other native attributes we don't explicitly accept should happen - pattern, inputmode, etc could be useful and currently not possible to use.

**Comment by jcfranco** [comment]
@caripizza Added! 
@macandcheese Agreed.

**Comment by driskull** [comment]
> Probably a larger discussion around other native attributes we don't explicitly accept should happen - pattern, inputmode, etc could be useful and currently not possible to use.

Agreed.

This is why I was thinking we could have a `input` slot on `calcite-input` where a user can just slot a native input with all these attributes. Then we could still have `calcite-input` with shadowRoot and maintain native form support as well.

**Comment by macandcheese** [comment]
That is Ionic's approach: https://github.com/ionic-team/ionic-framework/blob/2d6e45e8d11e3c1ebf00359fe99805c6994653a6/core/src/components/input/input.tsx#L45
https://github.com/ionic-team/ionic-framework/blob/2d6e45e8d11e3c1ebf00359fe99805c6994653a6/core/src/components/input/input.tsx#L408

Is this what that would look like?
```<calcite-input><input slot="input" readonly  inputmode="tel" /></calcite-input>```

**Comment by driskull** [comment]
I guess we should follow ionic 👍

**Comment by driskull** [comment]
@jcfranco should it be `readOnly` or `readonly`? `calcite-rating` uses camel casing, however the default HTMLInputElement does not.

**Comment by jcfranco** [comment]
@driskull For consistency, I think `readOnly`/`read-only` (prop/attr) makes sense.

**Comment by driskull** [comment]
Sounds good. PR open for this.

**Comment by caripizza** [comment]
Looks like `read-only` doesn't work on a numeric input's up/down arrows - I can still use the keyboard arrow keys to change the input's value. I can also use the Esc key to clear a numeric input's value. I tried this code with `1.0.0-next.265`:

```html
<calcite-label scale="s" for="s-number-1">
  Read-only clearable
  <calcite-input read-only clearable id="s-number-1" type="number" value="2"></calcite-input>
</calcite-label>

<calcite-label scale="s" for="s-number-2">
  Read-only number input
  <calcite-input read-only id="s-number-2" type="number"></calcite-input>
</calcite-label>
```

Should I create a new issue to make this new prop work on number inputs?

**Comment by driskull** [comment]
We can just use this issue @caripizza

**Comment by caripizza** [comment]
Got it @driskull, are you working on that already or should I take a stab at it?

**Comment by driskull** [comment]
I have something but you can take it over if you want :D

**Comment by driskull** [comment]
Installed fix

**Comment by caripizza** [comment]
Verified fixed on `1.0.0-next.266` using the same snippet from my above comment, closing this out.
